packages/aragon-wrapper/src/utils/transactions.js

```diff
@@ -39,44 +116,8 @@ export async function createDirectTransaction (sender, app, methodName, params,
     data: web3.eth.abi.encodeFunctionCall(methodABI, params)
   }
 
-  if (transactionOptions.token) {
-    const { address: tokenAddress, value: tokenValue, spender } = transactionOptions.token
-
-    const erc20ABI = getAbi('standard/ERC20')
-    const tokenContract = new web3.eth.Contract(erc20ABI, tokenAddress)
-    const balance = await tokenContract.methods.balanceOf(sender).call()
-
-    const tokenValueBN = toBN(tokenValue)
-
-    if (toBN(balance).lt(tokenValueBN)) {
-      throw new Error(`Balance too low. ${sender} balance of ${tokenAddress} token is ${balance} (attempting to send ${tokenValue})`)
-    }
-
-    const allowance = await tokenContract.methods.allowance(sender, destination).call()
-    const allowanceBN = toBN(allowance)
-
-    // If allowance is already greater than or equal to amount, there is no need to do an approve transaction
-    if (allowanceBN.lt(tokenValueBN)) {
-      if (allowanceBN.gt(toBN(0))) {
-        // TODO: Actually handle existing approvals (some tokens fail when the current allowance is not 0)
-        console.warn(`${sender} already approved ${destination}. In some tokens, approval will fail unless the allowance is reset to 0 before re-approving again.`)
-      }
-
-      // Approve the app (destination) unless an spender is passed to approve a different contract
-      const approveSpender = spender || destination
```

Comment:
Oops, this was actually done a bit too late; we should've used the `spender` when checking the initial allowance.